nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-client-service/src/main/java/org/apache/nifi/distributed/cache/client/CacheClientRequestHandler.java

```diff
@@ -64,6 +64,18 @@ public void channelReadComplete(ChannelHandlerContext ctx) throws IOException {
         }
     }
 
+    @Override
+    public void channelUnregistered(ChannelHandlerContext ctx) {
+        if (!inboundAdapter.isComplete()) {
+            channelPromise.setFailure(new IOException("channelUnregistered during request - " + ctx.channel().toString()));
```

Comment:
What do you think about changing the message it indicate incomplete processing status?
```suggestion
            channelPromise.setFailure(new IOException("Channel unregistered before processing completed: " + ctx.channel().toString()));
```